pgjdbc/src/test/java/org/postgresql/test/jdbc2/PreparedStatementTest.java

```diff
@@ -1093,6 +1095,46 @@ public void testSetObjectBigDecimalWithScale() throws SQLException {
   }
 
   @Test
+  public void testSetObjectNumberFallback() throws SQLException {
+    TestUtil.createTempTable(con, "number_fallback",
+            "n1 numeric");
+    PreparedStatement psinsert = con.prepareStatement("insert into number_fallback values(?)");
+    PreparedStatement psselect = con.prepareStatement("select n1 from number_fallback");
+    PreparedStatement pstruncate = con.prepareStatement("truncate table number_fallback");
+
+    // Test BigDecimal.
+    psinsert.setObject(1, new BigDecimal("733"));
+    psinsert.execute();
+
+    ResultSet rs = psselect.executeQuery();
+    assertTrue(rs.next());
+    assertEquals(new BigDecimal("733"), rs.getBigDecimal(1));
+    pstruncate.execute();
+
+    // Test BigInteger
```

Comment:
Can you please split the test method to several ones and use relevant assert message? (see https://github.com/pgjdbc/pgjdbc/blob/master/CONTRIBUTING.md#test )